providers/darwin/process_darwin.go

```diff
@@ -209,7 +214,7 @@ func kern_procargs(pid int, p *process) error {
 	}
 
 	// args
-	for i := 0; i < int(argc); i++ {
+	for len(lines) > 0 && len(p.args) < int(argc) {
 		p.args = append(p.args, string(lines[0]))
 		lines = lines[1:]
 	}
```

Comment:
If above, we make lines (requires a nil check on `data`, but that should probably happen anyway)
```
	lines := bytes.Split(buf.String(), "\x00")
	p.exe = string(lines[0])
	lines = lines[1:]
```
then this can become
```
p.args = make([]string, argc) // maybe min(argc, len(lines)) for precision? then drop last line here.
n := copy(p.args, lines)
p.args = p.args[:n]
```
with reduced allocations and work, and clearer semantics.

Reply:
Thanks for the feedback. PTAL